### PR TITLE
Update vs mac get started tutorials

### LINKED ISSUE #9490: Unclear that I had to generate a tasks.json and fill in launch.json
I ran into errors debugging because I needed to generate a tasks.json file, also i needed to fill out launch.json with the path to app.dll

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3343b35c-90c1-2e11-498d-ee1d4eb409f6
* Version Independent ID: f2eda261-d4c3-c90a-1294-dfe11e55c341
* Content: [Getting started with .NET Core on macOS](https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-on-macos)
* Content Source: [docs/core/tutorials/using-on-macos.md](https://github.com/dotnet/docs/blob/master/docs/core/tutorials/using-on-macos.md)
* Product: **dotnet-core**
* GitHub Login: @bleroy
* Microsoft Alias: **mairaw**

### LINKED ISSUE #12762: Please explain how to debug unit tests.
I see how to debug the app, but it doesn't mention how to debug unit tests.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 98d01d28-eba7-401c-5716-1361edfa928d
* Version Independent ID: f235603a-edcc-257a-901e-b0709cf81053
* Content: [Building a complete .NET Core solution on macOS using Visual Studio for Mac - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-on-mac-vs-full-solution#creating-a-test-project)
* Content Source: [docs/core/tutorials/using-on-mac-vs-full-solution.md](https://github.com/dotnet/docs/blob/master/docs/core/tutorials/using-on-mac-vs-full-solution.md)
* Product: **dotnet-core**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

### LINKED ISSUE #14290: Incorrect hot keys used
Since this is the macOS version of getting started a few changes need to happen.

Under the header Debug the Application, first paragraph it says to use F9 to insert a breakpoint.  On a mac users must use the combination fn+F9.  In the second paragraph, the hotkey combo listed to show the debug screen is Ctrl + Shift + D.  On a mac, it should be Cmd + Shift + D.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3343b35c-90c1-2e11-498d-ee1d4eb409f6
* Version Independent ID: f2eda261-d4c3-c90a-1294-dfe11e55c341
* Content: [Get started with .NET Core on macOS - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/tutorials/using-on-macos)
* Content Source: [docs/core/tutorials/using-on-macos.md](https://github.com/dotnet/docs/blob/master/docs/core/tutorials/using-on-macos.md)
* Product: **dotnet-core**
* GitHub Login: @bleroy
* Microsoft Alias: **mairaw**

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to #16046 

Two of the original steps are no longer needed due to new features in Visual Studio for Mac 2019. New screenshots were done for several of the images as well.

In addition, review and update (as needed) tutorials for VS Code / Mac and the initial Visual Studio for Mac tutorial. This meant also addressing these issues:

Fixes #12762
Fixes #9490
Fixes #14290

### CODE REVIEW

**Comment by Youssef1313** [comment]
I think the old picture was much better in quality

image

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 1]
```diff
@@ -19,27 +19,26 @@ This tutorial shows you how to create an application that accepts a search word
...
-- [.NET Core SDK 1.1 or later](https://dotnet.microsoft.com/download)
-- [Visual Studio 2017 for Mac](https://visualstudio.microsoft.com/vs/mac/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link)
+- [.NET Core SDK 3.1 or later](https://dotnet.microsoft.com/download)
+- [Visual Studio 2019 for Mac](https://visualstudio.microsoft.com/vs/mac/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link)
 
-For more information on prerequisites, see the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore30&pivots=os-macos). For the full system requirements of Visual Studio 2017 for Mac, see [Visual Studio 2017 for Mac Product Family System Requirements](/visualstudio/productinfo/vs2017-system-requirements-mac).
+For more information on prerequisites, see the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore30&pivots=os-macos). For the full system requirements of Visual Studio 2019 for Mac, see [Visual Studio 2019 for Mac Product Family System Requirements](/visualstudio/productinfo/vs2017-system-requirements-mac).
```
Comment (Youssef1313) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]:
```suggestion
For more information on prerequisites, see the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore31&pivots=os-macos). For the full system requirements of Visual Studio 2019 for Mac, see [Visual Studio 2019 for Mac Product Family System Requirements](/visualstudio/productinfo/vs2019-system-requirements-mac).
```

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 2]
```diff
@@ -19,27 +19,26 @@ This tutorial shows you how to create an application that accepts a search word
...
 
 ## Building a library
 
-1. On the Welcome screen, select **New Project**. In the **New Project** dialog under the **.NET Core** node, select the **.NET Standard Library** template. This creates a .NET Standard library that targets .NET Core as well as any other .NET implementation that supports version 2.0 of the [.NET Standard](../../standard/net-standard.md). Select **Next**.
+1. On the Welcome screen, select **New Project**. In the **New Project** dialog under the **.NET Core** node, select the **.NET Standard Library** template. This ccommand reates a .NET Standard library that targets .NET Core as well as any other .NET implementation that supports version 2.1 of the [.NET Standard](../../standard/net-standard.md). If you have multiple versions of the .NET Core SDK installed, you can choose different versions of .NET Standard for your library. Choose ".NET Standard 2.1" and select **Next**.
```
Comment (Youssef1313) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]:
```suggestion
1. On the Welcome screen, select **New Project**. In the **New Project** dialog under the **.NET Core** node, select the **.NET Standard Library** template. This command creates a .NET Standard library that targets .NET Core as well as any other .NET implementation that supports version 2.1 of the [.NET Standard](../../standard/net-standard.md). If you have multiple versions of the .NET Core SDK installed, you can choose different versions of .NET Standard for your library. Choose ".NET Standard 2.1" and select **Next**.
```

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 3]
```diff
@@ -141,41 +140,37 @@ Unit tests provide automated software testing during your development and publis
...
+1. ctrl-click the `WordCounterApp` project. Select **Start Debugging Project** from the context menu. When the app runs, enter the search word "cat" and "The dog chased the cat, but the cat escaped." for the string to search. When the `Console.WriteLine` statement is reached, program execution halts before the statement is executed. In the **Locals** tab, you can see the `searchWord`, `inputString`, `wordCount`, and `pluralChar` values.
 
    ![Visual Studio for Mac debugger program execution stopped](./media/using-on-mac-vs-full-solution/visual-studio-mac-debugger.png)
 
-1. In the **Immediate** pane, type "wordCount = 999;" and press Enter. This assigns a nonsense value of 999 to the `wordCount` variable showing that you can replace variable values while debugging.
+1. In the **Immediate** pane, type "wordCount = 999;" and press Enter. This command  assigns a nonsense value of 999 to the `wordCount` variable showing that you can replace variable values while debugging.
```
Comment (Youssef1313) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]:
```suggestion
1. In the **Immediate** pane, type "wordCount = 999;" and press Enter. This command assigns a nonsense value of 999 to the `wordCount` variable showing that you can replace variable values while debugging.
```

**Review by Youssef1313 — CHANGES_REQUESTED** [review]

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 4]
```diff
@@ -19,27 +19,26 @@ This tutorial shows you how to create an application that accepts a search word
...
+1. In the **Solution** pad, expand the `TextUtils` node to reveal the class file provided by the template, *Class1.cs*. Ctrl-click the file, select **Rename** from the context menu, and rename the file to *WordCount.cs*. Open the file and replace the contents with the following code:
 
    [!code-csharp[Main](../../../samples/core/tutorials/using-on-mac-vs-full-solution/WordCounter/TextUtils/WordCount.cs)]
 
-1. Save the file by using any of three different methods: use the keyboard shortcut <kbd>&#8984;</kbd>+<kbd>s</kbd>, select **File** > **Save** from the menu, or right-click on the file's tab and select **Save** from the contextual menu. The following image shows the IDE window:
+1. Save the file by using any of three different methods: use the keyboard shortcut <kbd>&#8984;</kbd>+<kbd>s</kbd>, select **File** > **Save** from the menu, or ctrl-click on the file's tab and select **Save** from the contextual menu. The following image shows the IDE window:
 
    ![Visual Studio for Mac IDE window with class library file and method](./media/using-on-mac-vs-full-solution/visual-studio-mac-editor.png)
```
Comment (tdykstra) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]: This image and the next two aren't showing up in staging.

**Review by tdykstra — COMMENTED** [review]
The title has WIP but reviews were requested so I did a cursory review.  Noted some broken images, and some that should have borders don't have them (https://review.docs.microsoft.com/en-us/help/contribute/contribute-how-to-create-screenshot?branch=master#gray-border-example). Otherwise LGTM.  I just read through the docs, didn't run the steps in VS/VSC.

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 2 above)
Comment (Youssef1313) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: @tdykstra , A typo.
`This ccommand reates` -> `This command creates`.

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 4 above)
Comment (BillWagner) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: That's now fixed. (occasionally they still timeout for me, but reloading the image works).

**Comment by BillWagner** [comment]
@tdykstra Do you have a list of which images need the grey border? I didn't see comments on that, other than your overall comment.

**Comment by tdykstra** [comment]
> Do you have a list of which images need the grey border?

It's a judgment call, so it's up to you.  I sent some additional comments by email.

**Comment by BillWagner** [comment]
This should be ready for final review @tdykstra @gewarren @mairaw

**Comment by mairaw** [comment]
I'll finish the two I'm working on and then review this

File: docs/core/tutorials/using-on-mac-vs.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ Visual Studio for Mac provides a full-featured Integrated Development Environmen
 
 ## Prerequisites
 
-See the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore30&pivots=os-macos) topic.
+See the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore30&pivots=os-macos) article.
```
Comment (Youssef1313) [inline docs/core/tutorials/using-on-mac-vs.md]: Is this article specific to core 3.0 ?
If not, I'd make the tab .NET Core 3.1, or maybe even remove the `tabs=` so it points to the latest version.

File: docs/core/tutorials/using-on-macos.md [hunk 6]
```diff
@@ -17,13 +17,13 @@ Install the [.NET Core SDK](https://dotnet.microsoft.com/download). The .NET Cor
 
 Install [Visual Studio Code](https://code.visualstudio.com). During the course of this article, you also install Visual Studio Code extensions that improve the .NET Core development experience.
 
-Install the Visual Studio Code C# extension by opening Visual Studio Code and pressing <kbd>F1</kbd> to open the Visual Studio Code palette. Type **ext install** to see the list of extensions. Select the C# extension. Restart Visual Studio Code to activate the extension. For more information, see the [Visual Studio Code C# Extension documentation](https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger.md).
+Install the Visual Studio Code C# extension by opening Visual Studio Code and pressing <kbd>Fn</kbd>+<kbd>F1</kbd> to open the Visual Studio Code palette. Type **ext install** to see the list of extensions. Select the C# extension. Restart Visual Studio Code to activate the extension. For more information, see the [Visual Studio Code C# Extension documentation](https://github.com/OmniSharp/omnisharp-vscode/blob/master/debugger.md).
```
Comment (Youssef1313) [inline docs/core/tutorials/using-on-macos.md]: I think the VSCode docs don't include the `Fn` in any keyboard shortcut.

Comment (BillWagner) [reply docs/core/tutorials/using-on-macos.md]: I know, but for this article, it was specifically reported in#14290

File: docs/core/tutorials/using-on-macos.md [hunk 7]
```diff
@@ -17,13 +17,13 @@ Install the [.NET Core SDK](https://dotnet.microsoft.com/download). The .NET Cor
...
 
-Start Visual Studio Code. Press <kbd>Ctrl</kbd>+<kbd>\`</kbd> (the backquote or backtick character) or select **View > Integrated Terminal** from the menu to open an embedded terminal in Visual Studio Code. You can still open an external shell with the Explorer **Open in Command Prompt** command (**Open in Terminal** on Mac or Linux) if you prefer to work outside of Visual Studio Code.
+Start Visual Studio Code. Press <kbd>Fn</kbd>+<kbd>Ctrl</kbd>+<kbd>\`</kbd> (the backquote or backtick character) or select **View > Terminal** from the menu to open an embedded terminal in Visual Studio Code. You can still open an external shell with the Explorer **Open in Command Prompt** command (**Open in Terminal** on Mac or Linux) if you prefer to work outside of Visual Studio Code.
```
Comment (Youssef1313) [inline docs/core/tutorials/using-on-macos.md]: I'm not using a Mac, so I might be mistaken. but `Fn` in this specific sentence seems too strange. There is no F1, F2, etc key after it.

Comment (BillWagner) [reply docs/core/tutorials/using-on-macos.md]: Good catch. Copy / paste got the better of me.

File: docs/core/toc.yml [hunk 8]
```diff
@@ -85,7 +85,7 @@
       href: tutorials/cli-templates-create-project-template.md
     - name: 3 - Create a template pack
       href: tutorials/cli-templates-create-template-pack.md
-  - name: Get started with .NET Core on macOS
+  - name: Get started with .NET Core on macOS using Visual Studio Code
```
Comment (mairaw) [inline docs/core/toc.yml]: The change looks good but I think we should shuffle some topics around in the TOC later

This one probably belongs in the Get Started node. The ones I'm working on are a sequence of tutorials that probably should be in the tutorials node with a well-defined sequence

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 9]
```diff
@@ -19,63 +19,70 @@ This tutorial shows you how to create an application that accepts a search word
...
-- [.NET Core SDK 1.1 or later](https://dotnet.microsoft.com/download)
-- [Visual Studio 2017 for Mac](https://visualstudio.microsoft.com/vs/mac/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link)
+- [.NET Core SDK 3.1 or later](https://dotnet.microsoft.com/download)
+- [Visual Studio 2019 for Mac](https://visualstudio.microsoft.com/vs/mac/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link)
 
-For more information on prerequisites, see the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore30&pivots=os-macos). For the full system requirements of Visual Studio 2017 for Mac, see [Visual Studio 2017 for Mac Product Family System Requirements](/visualstudio/productinfo/vs2017-system-requirements-mac).
+For more information on prerequisites, see the [.NET Core dependencies and requirements](../install/dependencies.md?tabs=netcore31&pivots=os-macos). For the full system requirements of Visual Studio 2019 for Mac, see [Visual Studio 2019 for Mac Product Family System Requirements](/visualstudio/productinfo/vs2019-system-requirements-mac).
```
Comment (mairaw) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]: I've been removing tabs from the links so it goes to whatever is latest

```suggestion
For more information on prerequisites, see the [.NET Core dependencies and requirements](../install/dependencies.md?pivots=os-macos). For the full system requirements of Visual Studio 2019 for Mac, see [Visual Studio 2019 for Mac Product Family System Requirements](/visualstudio/productinfo/vs2019-system-requirements-mac).
```

File: docs/core/toc.yml (hunk 8 above)
Comment (BillWagner) [reply docs/core/toc.yml]: Do you want me to change it now?

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 10]
```diff
@@ -19,63 +19,70 @@ This tutorial shows you how to create an application that accepts a search word
...
 
 ## Building a library
 
-1. On the Welcome screen, select **New Project**. In the **New Project** dialog under the **.NET Core** node, select the **.NET Standard Library** template. This creates a .NET Standard library that targets .NET Core as well as any other .NET implementation that supports version 2.0 of the [.NET Standard](../../standard/net-standard.md). Select **Next**.
+1. On the Welcome screen, select **New Project**. In the **New Project** dialog under the **.NET Core** node, select the **.NET Standard Library** template. This command creates a .NET Standard library that targets .NET Core as well as any other .NET implementation that supports version 2.1 of the [.NET Standard](../../standard/net-standard.md). If you have multiple versions of the .NET Core SDK installed, you can choose different versions of .NET Standard for your library. Choose ".NET Standard 2.1" and select **Next**.
```
Comment (mairaw) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]: We've been calling these start window for the VS topics. Should it be consistent?

```suggestion
1. On the start window, select **New Project**. In the **New Project** dialog under the **.NET Core** node, select the **.NET Standard Library** template. This command creates a .NET Standard library that targets .NET Core as well as any other .NET implementation that supports version 2.1 of the [.NET Standard](../../standard/net-standard.md). If you have multiple versions of the .NET Core SDK installed, you can choose different versions of .NET Standard for your library. Choose ".NET Standard 2.1" and select **Next**.
```

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 11]
```diff
@@ -1,13 +1,13 @@
 ---
 title: Building a complete .NET Core solution on macOS using Visual Studio for Mac
```
Comment (mairaw) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]:
```suggestion
title: Build a complete .NET Core solution using Visual Studio for Mac
```

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 12]
```diff
@@ -1,13 +1,13 @@
...
-description: This topic walks you through building a .NET Core solution that includes a reusable library and unit testing.
+description: This article walks you through building a .NET Core solution that includes a reusable library and unit testing.
 author: mairaw
-ms.date: 06/12/2017
+ms.date: 12/19/2019
 ms.custom: seodec18
 ---
 # Building a complete .NET Core solution on macOS using Visual Studio for Mac
```
Comment (mairaw) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]:
```suggestion
# Build a complete .NET Core solution on macOS using Visual Studio for Mac
```

File: docs/core/tutorials/using-on-mac-vs-full-solution.md [hunk 13]
```diff
@@ -19,63 +19,70 @@ This tutorial shows you how to create an application that accepts a search word
...
 
 1. Name the project "TextUtils" (a short name for "Text Utilities") and the solution "WordCounter". Leave **Create a project directory within the solution directory** checked. Select **Create**.
 
-   ![Visual Studio for Mac New project dialog options](./media/using-on-mac-vs-full-solution/visual-studio-mac-new-project-options.png)
+   > [!div class="mx-imgBorder"]
+   > ![Visual Studio for Mac New project dialog options](./media/using-on-mac-vs-full-solution/visual-studio-mac-new-project-options.png)
 
-1. In the **Solution** sidebar, expand the `TextUtils` node to reveal the class file provided by the template, *Class1.cs*. Right-click the file, select **Rename** from the context menu, and rename the file to *WordCount.cs*. Open the file and replace the contents with the following code:
+1. In the **Solution** pad, expand the `TextUtils` node to reveal the class file provided by the template, *Class1.cs*. Ctrl-click the file, select **Rename** from the context menu, and rename the file to *WordCount.cs*. Open the file and replace the contents with the following code:
```
Comment (mairaw) [inline docs/core/tutorials/using-on-mac-vs-full-solution.md]: why change from right-click?

File: docs/core/tutorials/using-on-mac-vs.md [hunk 14]
```diff
@@ -2,12 +2,12 @@
 title: Get started with .NET Core on macOS using Visual Studio for Mac
```
Comment (mairaw) [inline docs/core/tutorials/using-on-mac-vs.md]:
```suggestion
title: Get started with .NET Core using Visual Studio for Mac
```

**Review by mairaw — APPROVED** [review]
LGTM. Left some comments for you to consider.

File: docs/core/toc.yml (hunk 8 above)
Comment (mairaw) [reply docs/core/toc.yml]: If we can, sure! Otherwise, we can do it later.

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 13 above)
Comment (BillWagner) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: There's only one button on a Mac mouse.

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 12 above)
Comment (gewarren) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: Actually "Create" might be a better verb, so as not to confuse with "compile".

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 11 above)
Comment (gewarren) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: "Create" might be a better verb.

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 10 above)
Comment (gewarren) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: Actually I don't even know what this screen looks like on VS for Mac. Is it the same start window as VS on Windows?

Comment (gewarren) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: I see they do call it the start window, although it looks slightly different.

https://docs.microsoft.com/en-us/visualstudio/mac/ide-tour?view=vsmac-2019#getting-started

File: docs/core/tutorials/using-on-mac-vs-full-solution.md (hunk 11 above)
Comment (BillWagner) [reply docs/core/tutorials/using-on-mac-vs-full-solution.md]: You have a point. I looked at the other articles, and they use "build". I'll merge this as is, and we can update all titles later if we want.

**Pull request merged**